thorlcr/thorutil/thmem.cpp

```diff
@@ -646,7 +662,17 @@ void CThorExpandingRowArray::compact()
         }
     }
     numRows = freeFinger-rows;
-    // JCSMORE - this may be a good time to call IRowManager::compactRows
+    memsize_t numEmptiedPages = rowManager->compactRows(numRows, rows);
+
+#ifdef _DEBUG
+    ActPrintLog(&activity, "CThorExpandingRowArray::compact(): compactRows freed %"I64F"d pages", (unsigned __int64)numEmptiedPages);
+#endif
+
+    if (maxSpillCost)
```

Comment:
Looking at roxiemem, resizeRow() , I was under the impression that shrinking could allocate in order to shrink... (?)

So I pass in maxSpillCost to control when/what priority I want it to allow spilling at.
Passing 0 indicates that I don't want it to spill under any circumstances to allocate.